include/tim/vx/graph.h

```diff
@@ -57,12 +57,24 @@ class Graph {
   template <typename OpType, typename... Params>
   std::shared_ptr<OpType> CreateOperation(Params... parameters) {
     auto op = std::make_shared<OpType>(this, parameters...);
-    opVector.push_back(op);
+    op_vector_.push_back(op);
     return op;
   }
 
-private:
-  std::vector<std::shared_ptr<tim::vx::Operation>> opVector;
+  virtual const std::vector<std::shared_ptr<Tensor>> InputsTensor() const = 0;
+  virtual const std::vector<std::shared_ptr<Tensor>> OutputsTensor() const = 0;
+
+  virtual void UpdateTensorConsumersMap(
```

Comment:
shall we put this into private/protected section?

Reply:
We cannot put the API into private/protect section, because it's used out of the Graph class.